src/NUnitFramework/framework/Internal/Results/TestSuiteResult.cs

```diff
@@ -112,46 +194,66 @@ public override IList<ITestResult> Children
         /// <param name="result">The result to be added</param>
         public virtual void AddResult(ITestResult result)
         {
-            Children.Add(result);
+#if PARALLEL
+            var childrenAsConcurrentQueue = Children as ConcurrentQueue<ITestResult>;
+            if (childrenAsConcurrentQueue != null)
+                childrenAsConcurrentQueue.Enqueue(result);
+            else
+#endif
+            {
+                var childrenAsIList = Children as IList<ITestResult>;
+                if (childrenAsIList != null)
+                    childrenAsIList.Add(result);
+                else
+                    throw new NotSupportedException("cannot add results to Children");
```

Comment:
I realize that TestCaseResult overrides TestResult, and not TestSuiteResult.  I was only using it as an example.

The alternative is to make TestSuiteResult a sealed class, which would prevent derivation, and allow the use of _children like you suggested.